src/libpriv/rpmostree-postprocess.c

```diff
@@ -1476,8 +1476,9 @@ mutate_os_release (const char    *contents,
       g_string_append_printf (new_contents, "%s\n", line);
     }
 
-  /* add a bona fide ostree entry */
-  g_string_append_printf (new_contents, "OSTREE_VERSION=%s\n", next_version);
+  /* Add a bona fide ostree entry. Quote it as a precaution */
+  g_autofree char *quoted_version = g_shell_quote(next_version);
```

Comment:
So, in theory the other values we sub in higher up could suffer from this too. `VERSION` and `PRETTY_NAME` *are* quoted in at least e.g. Fedora, but using double quotes. I'd agree with @cgwalters though that in general if you're working on Fedora content, the `version` metadata key should be a "numeric-ish" value (another way to say this is that it should be the same "type" as what `30` means in `VERSION` and `PRETTY_NAME` since that's the string that gets substituted out in those).

Reply:
> So, in theory the other values we sub in higher up could suffer from this too. 

Indeed, I've juste realized that.

I think the idea here is to handle basic whitespace cases which is why I started with a really simple patch. There are no security implications as this string is provided at image build time and unlikely to contain strange characters.

> version metadata key should be a "numeric-ish" value

For the actual content in `VERSION`, reading https://www.freedesktop.org/software/systemd/man/os-release.html and looking at a classic Fedora Workstation install (`VERSION="30 (Thirty)"`), it feels like I could keep the `Kinoite` part but move it after the version number.

Note that this does not impact `VERSION_ID=30` in my builds.

I did this change originally to make it simpler to figure out which variant I was booting with GRUB during testing.